src/openrct2/ride/Vehicle.cpp

```diff
@@ -1520,13 +1550,11 @@ void Vehicle::UpdateMeasurements()
             curRide->testing_flags &= ~RIDE_TESTING_POWERED_LIFT;
         }
 
-        if (curRide->type == RIDE_TYPE_WATER_COASTER)
-        {
-            if (trackElemType >= TrackElemType::FlatCovered && trackElemType <= TrackElemType::RightQuarterTurn3TilesCovered)
-            {
-                curRide->special_track_elements |= RIDE_ELEMENT_TUNNEL_SPLASH_OR_RAPIDS;
-            }
-        }
+        const auto& ted = GetTrackElementDescriptor(trackElemType);
+        uint16_t trackFlags = ted.Flags;
```

Comment:
This could have been left down there, closer to its usage, as they are not needed for the 2 lines below